src/apps/DesktopAppPage.vala

```diff
@@ -400,5 +563,470 @@ namespace Ilia {
+                                if (in_arg) {
+                                    in_arg = false;
+                                    continue;
+                                }
+                                
+                                clean_exec += exec_line[i].to_string();
+                            }
+                            
+                            // Launch in terminal
+                            string terminal_cmd = "x-terminal-emulator -e " + clean_exec.strip();
+                            string[] args = {"sh", "-c", terminal_cmd};
+                            Process.spawn_async(null, args, null, SpawnFlags.SEARCH_PATH, null, null);
+                            hide_app_actions();
+                            session_controller.quit();
+                        } catch (Error e) {
+                            stderr.printf("Failed to launch in terminal: %s\n", e.message);
+                            hide_app_actions();
+                        }
+                        break;
+                        
+                    case "generic-shortcut":
+                        try {
+                            // Create desktop shortcut
+                            string home_dir = Environment.get_home_dir();
+                            string desktop_dir = Path.build_filename(home_dir, "Desktop");
+                            
+                            // Ensure desktop directory exists
+                            var dir = File.new_for_path(desktop_dir);
+                            if (!dir.query_exists()) {
+                                dir.make_directory_with_parents();
+                            }
+                            
+                            // Copy desktop file to Desktop folder
+                            var source = File.new_for_path(desktop_file_path);
+                            var dest = File.new_for_path(Path.build_filename(desktop_dir, Path.get_basename(desktop_file_path)));
+                            
+                            source.copy(dest, FileCopyFlags.OVERWRITE);
+                            
+                            // Make it executable
+                            FileUtils.chmod(dest.get_path(), 0755);
+                            
+                            hide_app_actions();
+                            
+                            // Print status message to console
+                            stdout.printf("Desktop shortcut created\n");
+                        } catch (Error e) {
+                            stderr.printf("Failed to create desktop shortcut: %s\n", e.message);
+                            hide_app_actions();
+                        }
+                        break;
+                        
+                    case "generic-folder":
+                        try {
+                            // Open the folder containing the desktop file
+                            string folder_path = Path.get_dirname(desktop_file_path);
```

Comment:
I don't think most end users want to navigate to the desktop file. You can remove this case as well.